task5/src/main/test/com/clouway/adapter/http/TransactionHistoryTest.java

```diff
@@ -0,0 +1,76 @@
+package com.clouway.adapter.http;
+
+import com.clouway.core.FundsRepository;
+import org.jmock.Expectations;
+import org.jmock.auto.Mock;
+import org.jmock.integration.junit4.JUnitRuleMockery;
+import org.junit.Rule;
+import org.junit.Test;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintWriter;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * Created by Kristiyan Petkov  <kristiqn.l.petkov@gmail.com> on 13.06.16.
+ */
+public class TransactionHistoryTest {
+  @Rule
+  public JUnitRuleMockery context = new JUnitRuleMockery();
+
+  @Mock
+  FundsRepository fundsRepository;
+
+  @Mock
+  HttpServletRequest request;
+
+  @Mock
+  HttpServletResponse response;
+
+  @Test
+  public void firstPage() throws Exception {
```

Comment:
Sometimes when you enter the pages there is `required parameters`. If this param is missing situation must be handeled - can be with message or in your case just load the first page records. I meant that will be good to have such a check.